src/jit/codegenarm64.cpp

```diff
@@ -3167,15 +3171,48 @@ CodeGen::genCodeForTreeNode(GenTreePtr treeNode)
                 varNum = compiler->lvaOutgoingArgSpaceVar;
             }
 
-            if (data->isContained())
+            // Do we have a TYP_STRUCT argument, if so it must be a 16-byte pass-by-value struct
+            if (targetType == TYP_STRUCT)
             {
-                getEmitter()->emitIns_S_I(ins_Store(targetType), emitTypeSize(targetType), varNum,
-                                          argOffset, (int) data->AsIntConCommon()->IconValue());
+                // We will use two store instructions that each write a register sized value
+
+                // We must have a multi-reg struct that takes two slots
+                assert(curArgTabEntry->numSlots == 2);
+                assert(!data->isContained());  // Impossible to have a contained 16-byte operand
+
+                // We will need to determine the GC type to use for each of the stores
+                // We obtain the gcPtrs values by examining op1 using getStructGcPtrsFromOp()
+
+                BYTE gcPtrs[2] = {TYPE_GC_NONE, TYPE_GC_NONE};
+
+                compiler->getStructGcPtrsFromOp(data, &gcPtrs[0]);
+
+                var_types type0 = compiler->getJitGCType(gcPtrs[0]);
+                var_types type1 = compiler->getJitGCType(gcPtrs[1]);
+
+                genConsumeReg(data); 
```

Comment:
Here too lowerarm64.cpp seems to be not specifying reg requirements.  I see internalcount set on putarg_stk but don't see that utilized here.